src/ResourceManagement/Resource/Microsoft.Azure.Management.V2.Resource/Core/CreatableUpdatableResourcesRoot.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Management.V2.Resource.Core.ResourceActions;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Management.V2.Resource.Core
+{
+    /// <summary>
+    /// Implementation of <see cref="ICreatableUpdatableResourcesRoot{IFluentResourceT}">
+    /// </summary>
+    /// <typeparam name="IFluentResourceT">the type of resources in the batch</typeparam>
+    /// <typeparam name="InnerResourceT">the type of inner resource that IFluentResourceT wraps</typeparam>
+    internal class CreatableUpdatableResourcesRoot<IFluentResourceT, InnerResourceT> :
+        CreatableUpdatable<ICreatableUpdatableResourcesRoot<IFluentResourceT>,
+            InnerResourceT,
+            CreatableUpdatableResourcesRoot<IFluentResourceT, InnerResourceT>,
+            IResource,
+            object>,
```

Comment:
we could technically but as i mentioned in the previous comment we are not using the last generic arg and class is internal. If we want to do that we endup passing additional argument to `GroupableResources` from all collections, so at this point i would suggest going with this `object` and keep the collections impl clean. We can revise this root resource dummy logic in the next phase.